lib/ssoe/service.rb

```diff
@@ -56,5 +55,61 @@ def build_message(credential_method, credential_id, user, address)
         zipcode: address.zipcode
       ).perform
     end
+
+    def parse_response(response_body)
+      parsed = Hash.from_xml(response_body)
+
+      if parsed.dig('Envelope', 'Body', 'getSSOeTraitsByCSPIDResponse', 'icn')
```

Comment:
Do we need this `if` block and two digs for the value? `parsed.dig` for the `icn` [should just return `nil` if it's not found](https://docs.ruby-lang.org/en/3.3/dig_methods_rdoc.html) so you could do

```ruby
icn = parsed.dig('Envelope', 'Body', 'getSSOeTraitsByCSPIDResponse', 'icn')
return { success: true, icn: } if icn.present?
```